plugins/modules/mso_schema_site_anp_epg_bulk_staticport.py

```diff
@@ -374,95 +375,89 @@ def main():
+                pod = static_port.get("pod") or module_pod
+                leaf = static_port.get("leaf") or module_leaf
+                fex = static_port.get("fex") or module_fex
+                path = static_port.get("path") or module_path  # Note :path has to be diffent in each leaf for every static port in the list.
+                vlan = static_port.get("vlan") or module_vlan
+                primary_micro_segment_vlan = static_port.get("primary_micro_segment_vlan") or module_primary_micro_segment_vlan
+                deployment_immediacy = static_port.get("deployment_immediacy") or module_deployment_immediacy
+                mode = static_port.get("mode") or module_mode
+
+                required_dict = {"pod": pod, "leaf": leaf, "path": path, "vlan": vlan}
+                if None in required_dict.values():
+                    res = []
+                    for i in required_dict:
+                        if required_dict[i] is None:
+                            res.append(i)
+                    mso.fail_json(msg="state is present but all of the following are missing: {0}.".format(", ".join(res)))
+                else:
+                    if path_type == "port" and fex is not None:
+                        # Select port path for fex if fex param is used
+                        portpath = "topology/{0}/paths-{1}/extpaths-{2}/pathep-[{3}]".format(pod, leaf, fex, path)
+                    elif path_type == "vpc":
+                        portpath = "topology/{0}/protpaths-{1}/pathep-[{2}]".format(pod, leaf, path)
+                    else:
+                        portpath = "topology/{0}/paths-{1}/pathep-[{2}]".format(pod, leaf, path)
+
+                new_leaf = dict(
+                    deploymentImmediacy=deployment_immediacy,
+                    mode=mode,
+                    path=portpath,
+                    portEncapVlan=vlan,
+                    type=path_type,
+                )
+
+                if primary_micro_segment_vlan:
+                    new_leaf.update(microSegVlan=primary_micro_segment_vlan)
+
+                # validate and append staticports to staticport_list if path variable is different
+                if portpath in unique_paths:
+                    mso.fail_json(msg="Each leaf in a pod of a static port should have an unique path.")
+                else:
+                    unique_paths.append(portpath)
+                    staticport_list.append(new_leaf)
+
+        # If payload is empty, anp and EPG already exist at site level
+        if not payload:
+            payload = staticport_list
+        else:
+            # If anp already exists at site level
+            if "anpRef" not in payload:
+                payload["staticPorts"] = staticport_list
+            else:
+                payload["epgs"][0]["staticPorts"] = staticport_list
+
+        # mso.sanitize(payload, collate=True)
+        mso.proposed = deepcopy(staticport_list)
```

Comment:
Why do you do the deepcopy() here?

Reply:
when mso.sanitize is used it gives the below error, due to the datastructure in staticport_list

module_utils/mso.py", line 1094, in sanitize
TypeError: list indices must be integers or slices, not str.

Also, since the static_port list and payload are not modified further deepcopy() is removed